### DISCUSSION TITLE
Gallery/Camera randomly not opening

### DISCUSSION THREAD

**Discussion by msegers** [react-native-image-picker/react-native-image-picker#1778] [Q&A]
Hey people, 
I was wondering if anyone can push me in the right direction, we have been running into random issues that our customers can not open the `ImagePicker` gallery/camera. I've had it a few times during development when we were still running version 2.3.x. So we upgraded to the latest, which was `4.0.5` at that time I believe. Anyway we still have random issues with the picker not opening.

So this happens right when we start, the permission checks are in place I had a colleague who had the rando issue occuring on his iPhone XR. He had the correct permissions, and it SOMETIMES didn't work so the permissions seem to be fine.

we basically try to open the library e.g. as follows:

```
return launchImageLibrary(
      {
        mediaType: "photo",
        quality: 0.8,
        includeBase64: true
      },
      res => {
        if (res.errorCode) {
          onError(res);
        }
        return cb(res);
      }
    );
```
However nothing happens, no logging and also no errors. We have our `Sentry` setup for catching thrown errors and alike, the code above is also wrapped within a try-catch statement (for the permission check) but that's probably ignored as we do not await anything, but I would expect errors to occur within the callback.

I was hoping someone could point me in a direction I havent looked in, the worst part is that the issue seems to be mostly random.

**Comment by Johan-dutoit** [comment]
Not really related (or I don't think so anyway), but do you need the permission checks?  4.x.x removes the need to request permissions (for most scenarios anyway).

**Reply by msegers** [reply]
I read something like that, that would be interesting to look into, but that won't be the solution to the problem indeed.

**Comment by Johan-dutoit** [comment]
These types of issues are quite hard to solve, especially since it's seemingly random and doesn't have clear reproducible steps.

I have not personally had these issues, which I know doesn't help, but unfortunately doesn't help me provide a solution.
If you can somehow identify what causes it i.e. multiple open triggers (using pressing button twice fast), etc. then perhaps we can find out what the cause it.

**Reply by msegers** [reply]
Yes, If I get more info I will report back.
Do you know anything about no errors being triggered though? any chance it is swallowed by the native code?
